sources/api/ecs-settings-applier/src/main.rs

```diff
@@ -0,0 +1,150 @@
+#[macro_use]
+extern crate log;
+
+use std::{env, process};
+use std::fs;
+use std::path::{Path, PathBuf};
+use serde::{Serialize};
+use snafu::{ResultExt};
+use error::SettingsApplierError;
+
+const DEFAULT_API_SOCKET: &str = "/run/api.sock";
+const DEFAULT_ECS_CONFIG_PATH: &str = "/etc/ecs/ecs.config.json";
+const VARIANT_ATTRIBUTE_NAME: &str = "bottlerocket.variant";
+const VERSION_ATTRIBUTE_NAME: &str = "bottlerocket.version";
+
+
+#[derive(Serialize, Debug)]
+#[serde(rename_all="PascalCase")]
+struct ECSConfig {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    cluster: Option<String>,
+
+    #[serde(skip_serializing_if = "std::collections::HashMap::is_empty")]
+    instance_attributes: std::collections::HashMap<String, String>,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    privileged_disabled: Option<bool>,
+}
+
+// Returning a Result from main makes it print a Debug representation of the error, but with Snafu
+// we have nice Display representations of the error, so we wrap "main" (run) and print any error.
+// https://github.com/shepmaster/snafu/issues/110
+fn main() {
+    if let Err(e) = run() {
+        eprintln!("{}", e);
+        process::exit(1);
+    }
+}
+
+fn run() -> Result<()> {
+    let args = parse_args(env::args());
+
+    // Get all settings values for config file templates
+    debug!("Requesting settings values");
+    let settings = schnauzer::get_settings(&args.socket_path).context(error::SettingsError)?;
+
+    debug!("settings = {:#?}", settings.settings);
+    let ecs = settings.settings.and_then(|s| s.ecs);
```

Comment:
Maybe it can be ref-ified here to eliminate the subsequent .as_ref() calls? (Not sure).

```suggestion
    let ecs = settings.settings.and_then(|s| s.ecs).as_ref();
```

I was also trying to see if there was some simplified code path that could eliminate the profusion of `if_then`s, but I couldn't see it.

Sometimes you can do, like

```rust
let ecs = if let Some(x) = ecs {
    x
} else {
   // do something simple and bail
  return Ok(());
}

// lots of code where ecs is no longer an Option
```

Maybe this doesn't work for you though since os.settings is also an option.

Reply:
P.S. - not a blocker for me.

Reply:
@webern Thanks! Going to take a look and see if I can take your suggestions.  I tried a bunch of things with `as_ref()` and `if_then` when I was writing this, but I am very much a Rust newbie.

Reply:
```
error[E0716]: temporary value dropped while borrowed
  --> api/ecs-settings-applier/src/main.rs:48:15
   |
48 |     let ecs = settings.settings.and_then(|s| s.ecs).as_ref();
   |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^         - temporary value is freed at the end of this statement
   |               |
   |               creates a temporary which is freed while still in use
49 |     let cluster = ecs.and_then(|s| s.cluster.as_ref());
   |                   --- borrow later used here
```

I was able to use your `let` suggestion though; see the latest push for a commit making that change.